src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -476,10 +476,24 @@ public function __clone()
      */
     public function __toString()
     {
+        $protocol = '';
+        try {
+            $protocol = $this->server->get('SERVER_PROTOCOL');
+        } catch (\InvalidArgumentException $mallformedPath) {
```

Comment:
Emm, why? Code shall be self-explaining.

I could then rename $protocol to $p - it's clear from context what happens...